api-report/tree.api.md

```diff
@@ -4,9 +4,34 @@
 
 ```ts
 
+import { Serializable } from '@fluidframework/datastore-definitions';
+
+// @public
+export interface ITreeCursor {
+    down(key: TreeKey, index: number): TreeNavigationResult;
+    keys: Iterable<TreeKey>;
+    length(key: TreeKey): number;
+    type: TreeType;
+    up(): TreeNavigationResult;
+    value: undefined | Serializable;
+}
+
 // @public (undocumented)
-export const _ = "Force index.d.ts generation from ApiExtractor.";
+export type TreeKey = (number | string) & {
+    readonly TreeKey: symbol;
+};
 
+// @public (undocumented)
+export const enum TreeNavigationResult {
+    NotFound = -1,
+    Ok = 1,
+    Pending = 0
+}
+
+// @public (undocumented)
+export type TreeType = (number | string) & {
```

Comment:
This seems to omit the ideas of a) Definition/whatever-we-call-it, i.e. a run-time identity that will typically map directly to a given schema, b) identity, and c) treatment of CodePoints as the true text primitives, i.e. String being a schema that allows one to conveniently obtain a string.

Reply:
'TreeType' is intended to be Definition.  (During the review of the data model spec, we felt that 'type' was more accessible to developers than 'definition.)

There was a placeholder 'TreeId' type in an earlier iteration, but I removed it because it wasn't currently used by the ITreeCursor interface (and I know Taylor has a hierarchy of related Id types from Id compression.)

With respect to (c), perhaps that's best addressed by the proposal to add TreeType specific accessors to do things like iterate through a subtree as segments of homogenous types?